src/otx/v2/adapters/torch/lightning/engine.py

```diff
@@ -137,17 +137,45 @@ def _update_logger(
 
     def _update_callbacks(
         self,
-        callbacks: list[pl.Callback] | pl.Callback | None = None,
+        callbacks: list[pl.Callback] | pl.Callback | DictConfig | None = None,
+        mode: str | None = None,
     ) -> list[pl.Callback] | pl.Callback | None:
         """Update the list of callbacks to be executed during training and validation.
 
         Args:
-            callbacks(list[pl.Callback] | pl.Callback | None): Input of callbacks
+            callbacks(list[pl.Callback] | pl.Callback | DictConfig | None): Input of callbacks
+            mode(bool): Current Running mode status
 
         Returns:
             list[pl.Callback] | pl.Callback | None: Updated callbacks.
         """
 
+    def _load_checkpoint(
```

Comment:
From a quick glance I think we can drop `_load_checkpoint`. In Anomalib we use a `LoadModelCheckpoint` callback because we had some issues with loading the checkpoint files using lightning. The recommended method to load checkpoints in lightning is to pass the checkpoint path to the `fit`, `test`, `predict`, and `validate` method. So basically, `self.trainer.fit(model, datamodule, ckpt_path="/path/to/model.ckpt")`. I can see that's what you are doing in `fit` and `validate`. We can then just add a check something like
```python
def predict(self, ...):
    if checkpoint is None:
        checkpoint = self.latest_model.get("checkpoint")
    self.trainer.predict(model, ..., ckpt_path=checkpoint)
```
If checkpoint is None in `self.latest_model` then it won't break the execution. Optionally, we can also print a warning if both the parameter and the checkpoint path in self.latest_model is None.

And, since the trainer method takes care of calling `model.load_state_dict`, we can leave custom functionality to the model implementers.

As for your original question regarding loading the model for exporting and predicting, we can also do something like

```python
def export(self, model, checkpoint, ...):
    if checkpoint is None:
        checkpoint = self.latest_model.get("checkpoint")
    model = model.__class__.load_from_checkpoint(checkpoint)
```
I am not sure if it is a good programming practice to call `__class__` though. But here is where they mention how to load the checkpoint https://lightning.ai/docs/pytorch/stable/common/checkpointing_basic.html#lightningmodule-from-checkpoint

Reply:
Actually, the current flow seems to work fine, but I'll test your suggestion and report back. Thanks for the nice comment!

Reply:
I try to modify. However, the test, predict & export part is not working.
![image](https://github.com/openvinotoolkit/training_extensions/assets/38045080/999355b6-6e0d-4541-8d1f-73206d5c3afa)
(Note that the visual prompt works fine, but not with anomalib.)
So I think we should leave it as is for now.

Reply:
Yeah, let's not try to use this with anomalib. Since it is independent library let's treat lightning engine separate and we can address this differently in anomalib engine.